Source/Form/Gen3/Wild3.cpp

```diff
@@ -197,22 +197,37 @@ void Wild3::generatorEncounterIndexChanged(int index)
         {
         case Encounter::Grass:
             ui->filterGenerator->setEncounterSlots(12);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), false);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), false);
             break;
         case Encounter::RockSmash:
+            ui->filterGenerator->setEncounterSlots(5);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), true);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), true);
+            break;
         case Encounter::Surfing:
-        case Encounter::SuperRod:
             ui->filterGenerator->setEncounterSlots(5);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), false);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), true);
             break;
         case Encounter::OldRod:
             ui->filterGenerator->setEncounterSlots(2);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), true);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), true);
             break;
         case Encounter::GoodRod:
             ui->filterGenerator->setEncounterSlots(3);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), true);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), true);
+            break;
+        case Encounter::SuperRod:
+            ui->filterGenerator->setEncounterSlots(5);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), true);
+            ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), true);
             break;
         default:
             break;
         }
-
         updateEncounterGenerator();
```

Comment:
I think it will probably be cleaner to do it all below the switch statement. It also matches how I have done it in other places (for example Wild4.cpp)

```
bool magnet = encounter == Encounter::Grass;
bool static = encounter == Encounter::Grass || encounter == Encounter::Surfing;

ui->comboMenuGeneratorLead->hideAction(toInt(Lead::MagnetPull), !magnet);
ui->comboMenuGeneratorLead->hideAction(toInt(Lead::Static), !static);
```

Reply:
Changed as desired but with different variable names since static is a keyword